web.js

```diff
@@ -1,6 +1,278 @@
 const crypto = require('.')
+const errors = require('./lib/errors')
 
 // https://w3c.github.io/webcrypto/#Crypto-method-getRandomValues
 exports.getRandomValues = function getRandomValues(array) {
   return crypto.randomFillSync(array)
 }
+
+// https://w3c.github.io/webcrypto/#subtlecrypto-interface
+exports.SubtleCrypto = class SubtleCrypto {
+  // https://w3c.github.io/webcrypto/#SubtleCrypto-method-generateKey
+  // https://w3c.github.io/webcrypto/#hmac-operations-generate-key
+  async generateKey(algorithm, extractable, usages) {
+    let { name, length } = algorithm
+    const { hash } = algorithm
+
+    name = name.toUpperCase()
+
+    if (name !== 'HMAC') {
+      throw errors.UNSUPPORTED_ALGORITHM(`Unsupported algorithm name '${name}'`)
+    }
+
+    if (usages.length === 0) {
+      throw new SyntaxError('Usages argument cannot be empty')
+    }
+
+    for (const usage of usages) {
+      if (!['sign', 'verify'].includes(usage)) {
```

Comment:
```suggestion
      if (usage !== 'sign' && usage !== 'verify') {
```

At a certain number of cases a `switch` becomes more tidy and is readily optimized into a jump table.